src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__BIONIC__))
```

Comment:
I guess clang started to complain on this in version 3.8. We use 3.7 in the lab. I have verified I am getting the same problem with 3.9 as you do.

I have just realized why my suggestion didn't work. The problem is that "T" already is volatile, so casting to (T*) is a no-op.

What needs to be done instead is a bit tricky, since C++ doesn't have a construct to cast away volatile from a template parameter.

What we can do and I have verified to work is this. First, declare two helper template structs:

```C++
// Helper structures for casting away volatileness
template <typename T>
struct RemoveVolatile
{
    typedef T type;
};

template <typename T>
struct RemoveVolatile<volatile T>
{
    typedef T type;
};
```

Then change the problematic `__atomic_load` to:
```C++
 __atomic_load((T const *)pt, const_cast<typename RemoveVolatile<T>::type *>(&val), __ATOMIC_ACQUIRE);
```